src/transitions/Slide.js

```diff
@@ -163,6 +163,14 @@ class Slide extends React.Component<ProvidedProps & Props, State> {
     });
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps.direction !== this.props.direction) {
+      // We need to update the position of the drawer when the direction change and
+      // when it's hidden.
+      this.componentDidMount();
```

Comment:
Might have been better to extract the DidMount logic into a function and call that function from both of these methods. This might cause bugs in the future.

Reply:
I agre, also:
```diff
-to propery hide it.
+to properly hide it.
```

Reply:
Do you want to finish it? :)

Reply:
I'll take care of it. 😀